### PR TITLE
jdk23 Blocker.begin() in Object.wait is only called for virtual threads

### PR DESCRIPTION
**Pull request by pshipton**
Previous jdk23 builds were hanging, it's looking better with this change to match OpenJDK.
[8329593: Drop adjustments to target parallelism when virtual threads](https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/782/commits/e592d365bdcf133bc740531eba65f5bc01cedd0d)

Related to https://github.com/eclipse-openj9/openj9/pull/19374

Hanging builds https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-JDK23-with-System/5/

First PR build https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/517/

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins line endings check

**Comment by pshipton** [comment]
jenkins test sanity.openjdk zlinux jdknext

**Comment by pshipton** [comment]
jenkins test sanity zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#783

**Comment by pshipton** [comment]
> I don't see how this fixes anything

I can't explain it, but the tests seem to be working.
https://github.com/eclipse-openj9/openj9/pull/19417#issuecomment-2086316661 wasn't working for you earlier today.

**Comment by pshipton** [comment]
Running sanity.openjdk without the getAndClearInterrupt() change.
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/519/

**Comment by keithc-ca** [comment]
I think the sanity failure is due to https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/783; see https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/783#issuecomment-2087044691.

**Comment by pshipton** [comment]
@keithc-ca sanity.openjdk passed without the change to call getAndClearInterrupt(). I've updated this PR to remove it.
https://github.com/eclipse-openj9/openj9/pull/19417#issuecomment-2086995941

File: jcl/src/java.base/share/classes/java/lang/Object.java [hunk 1]
```diff
@@ -276,6 +276,9 @@ public final void wait(long time) throws InterruptedException {
 public final void wait(long time, int frac) throws InterruptedException {
 /*[IF JAVA_SPEC_VERSION >= 19]*/
 /*[IF JAVA_SPEC_VERSION >= 23]*/
+  if (!Thread.currentThread().isVirtual()) {
+    waitImpl(time, frac);
+  } else {
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/Object.java:281]: Please indent only with tabs (like the rest of this file).

**Comment by keithc-ca** [comment]
That method is getting hard to follow with all the preprocessor conditions, maybe it's time to split it by version?
```java
/*[IF JAVA_SPEC_VERSION < 19]*/
	waitImpl(time, frac);
/*[ELSEIF JAVA_SPEC_VERSION < 23]*/
	long blockerRC = Blocker.begin();
	try {
		waitImpl(time, frac);
	} finally {
		Blocker.end(blockerRC);
	}
/*[ELSE] JAVA_SPEC_VERSION < 23 */
	if (!Thread.currentThread().isVirtual()) {
		waitImpl(time, frac);
	} else {
		boolean blockerRC = Blocker.begin();
		try {
			waitImpl(time, frac);
		} finally {
			Blocker.end(blockerRC);
		}
	}
/*[ENDIF] JAVA_SPEC_VERSION < 19 */
```

**Comment by pshipton** [comment]
Updated it to split by versions.

**Comment by keithc-ca** [comment]
I still don't understand how this is anything other than a (small) optimization for non-virtual threads in jdk23 by hoisting up the call to `isVirtual()`, but I don't see any downsides to merging this.
